### DISCUSSION TITLE
Typing greek letters in notebook

### DISCUSSION THREAD

**Discussion by mapio** [microsoft/vscode-jupyter#9362] [Tips, Tricks and Workarounds]
I've long hoped for a solution to #806, #3315, and #3677 but apparently there will be no progress from the developers.

For all the users that would really like to have such feature, the [Greek Alphabet Snippets](https://marketplace.visualstudio.com/items?itemName=lucas-rezende.greek-alphabet-snippets) extension can be helpful.

**Comment by mapio** [comment]
See also https://github.com/microsoft/vscode-jupyter/issues/9363#issuecomment-1068232986 for another package that offers (if properly configured) even more completions.

**Comment by Atcold** [comment]
For anyone arriving from #806, #3315 or #9363 — I wrote an extension that does the actual Jupyter behaviour: type a LaTeX name, press <kbd>Tab</kbd>, get the character.

```
\beta   Tab → β
y\tilde Tab → ỹ
\bbR    Tab → ℝ
```

It uses IPython's own table — the 1300 entries from Julia's `latex_symbols.jl` that pass `("a" + char).isidentifier()` — so it completes exactly what the browser notebook completes, rather than a hand-written snippet list. Works in notebook cells and `.py` files; outside a `\name`, Tab still indents normally.

Not on the marketplace, so clone and build: https://github.com/Atcold/jupyter-unicode-vscode-ext — MIT.
